sa/sa.go

```diff
@@ -498,12 +499,25 @@ func (ssa *SQLStorageAuthority) NewOrderAndAuthzs(ctx context.Context, req *sapb
 		}
 
 		// Second, insert the new order.
-		order := &orderModel{
-			RegistrationID: req.NewOrder.RegistrationID,
-			Expires:        req.NewOrder.Expires.AsTime(),
-			Created:        ssa.clk.Now(),
+		var omv1 orderModelv1
+		var omv2 orderModelv2
```

Comment:
The diff in this function makes logical sense, but the result is difficult to read (so many feature flag conditionals!) and very error-prone (it would be _so easy_ to accidentally reference omv1 or omv2 outside of one of those conditionals, and the code would all compile but you'd only ever get the zero-value).

So instead, let's keep these isolated to the only place they're needed: tx.Insert().

Here's what I suggest: find all the values that you use below (omv.ID, omv.RegistrationID, omv.Expires, etc) and find other ways to provide them. For example, omv.RegistrationID can be replaced by req.NewOrder.RegistrationID.

The only one that will give you trouble is omv.ID, since that only exists after we've done the insert. For that, replace these two variable declarations with `var oid int64`, and then just after lines 512 and 519, assign `oid = omv[1|2].ID`. That way we can keep the feature-flag conditional isolated to _just this block_, and exfiltrate the new order ID for use later on in the function.